packages/chakra-components/src/components/Election/Questions/Form.tsx

```diff
@@ -22,87 +25,162 @@ export const useQuestionsForm = () => {
+        if (choices[0].includes(k.toString())) {
+          return 1
+        } else {
+          return 0
+        }
+      })
+      break
+    default:
+      throw new Error('Unknown or invalid election type')
+  }
+  return results
+}
+
+const useMultiElectionsProvider = ({
+  fmethods,
+  confirmContents,
+}: { fmethods: UseFormReturn } & QuestionsFormProviderProps) => {
   const { confirm } = useConfirm()
-  const { election, client, vote: bvote } = useElection()
+  const { client, isAbleToVote: rootIsAbleToVote, voted: rootVoted, election, vote } = useElection() // Root Election
+  // State to store on memory the loaded elections to pass it into confirm modal to show the info
+  const [electionsStates, setElectionsStates] = useState<ElectionStateStorage>({})
+  const [voting, setVoting] = useState<boolean>(false)
+
+  const voted = useMemo(
+    () => (electionsStates && Object.values(electionsStates).every(({ voted }) => voted) ? 'true' : null),
+    [electionsStates]
+  )
+
+  const isAbleToVote = useMemo(
+    () => electionsStates && Object.values(electionsStates).some(({ isAbleToVote }) => isAbleToVote),
+    [electionsStates]
+  )
+
+  // Add an election to the storage
+  const addElection = (electionState: SubElectionState) => {
+    setElectionsStates((prev) => ({
+      ...prev,
+      [(electionState.election as PublishedElection).id]: electionState,
+    }))
+  }
+
+  // Root election state to be added to the state storage
+  const rootElectionState: SubElectionState | null = useMemo(() => {
+    if (!election || !(election instanceof PublishedElection)) return null
+    return {
+      vote,
+      election,
+      isAbleToVote: rootIsAbleToVote,
+      voted: rootVoted,
+    }
+  }, [vote, election, rootIsAbleToVote, rootVoted])
 
-  const vote = async (values: FieldValues) => {
-    if (!election || !(election instanceof PublishedElection)) {
-      console.warn('vote attempt with no valid election defined')
+  // reset form if account gets disconnected
+  useEffect(() => {
+    if (typeof client.wallet !== 'undefined') return
+
+    setElectionsStates({})
+    fmethods.reset({
+      ...Object.values(electionsStates).reduce((acc, { election }) => ({ ...acc, [election.id]: '' }), {}),
+    })
+  }, [client, electionsStates, fmethods])
+
+  // Add the root election to the state to elections cache
+  useEffect(() => {
+    if (!rootElectionState || !rootElectionState.election) return
+    const actualState = electionsStates[rootElectionState.election.id]
+    if (rootElectionState.vote === actualState?.vote || rootElectionState.isAbleToVote === actualState?.isAbleToVote) {
+      return
+    }
+    addElection(rootElectionState)
+  }, [rootElectionState, electionsStates, election])
+
+  const voteAll = async (values: Record<string, FieldValues>) => {
+    if (!electionsStates || Object.keys(electionsStates).length === 0) {
+      console.warn('vote attempt with no valid elections not defined')
```

Comment:
Error should be `Vote attempt with no valid elections defined` (there's an incorrect double negation)

Reply:
d6e3168